### PR TITLE
DRILL-7850: Unable to load DrillConnectionImpl class from JDBC driver

### PR DESCRIPTION
**Pull request by vvysotskyi**
# [DRILL-7850](https://issues.apache.org/jira/browse/DRILL-7850): Unable to load DrillConnectionImpl class from JDBC driver

## Description
Please refer to https://github.com/jpype-project/jpype/issues/913#issuecomment-767789793 for the problem description.

## Documentation
NA

## Testing
Checked manually.

### CODE REVIEW

File: exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillConnectionImpl.java [hunk 1]
```diff
@@ -231,13 +217,7 @@ public DrillClient getClient() {
 
   @Override
   public void setAutoCommit(boolean autoCommit) throws SQLException {
```
Comment (luocooong) [inline exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillConnectionImpl.java:219]: Currently, `setAutoCommit()` override super class method but didn't do anything, is right?

File: exec/java-exec/src/test/java/org/apache/drill/exec/util/StoragePluginTestUtils.java [hunk 2]
```diff
@@ -46,13 +43,6 @@
   public static final String ROOT_SCHEMA = "root";
 
   public static final String DFS_TMP_SCHEMA = DFS_PLUGIN_NAME + "." + TMP_SCHEMA;
-  public static final String DFS_DEFAULT_SCHEMA = DFS_PLUGIN_NAME + "." + SchemaFactory.DEFAULT_WS_NAME;
```
Comment (luocooong) [inline exec/java-exec/src/test/java/org/apache/drill/exec/util/StoragePluginTestUtils.java:49]: Would these changes affect the test unit with marked `@ignore`?

**Review by luocooong — COMMENTED** [review]
@vvysotskyi Thanks for the PR. please take note of my comments. thanks.

File: exec/java-exec/src/test/java/org/apache/drill/exec/util/StoragePluginTestUtils.java (hunk 2 above)
Comment (vvysotskyi) [reply exec/java-exec/src/test/java/org/apache/drill/exec/util/StoragePluginTestUtils.java:49]: No, they shouldn't affect any tests. The logic responsible for creating workspaces for tests was updated and moved from `DrillConnectionImpl` to the `JdbcTestBase` class, and these constants aren't required anymore.

File: exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillConnectionImpl.java (hunk 1 above)
Comment (vvysotskyi) [reply exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillConnectionImpl.java:219]: Yes, it is correct. Drill doesn't support transactions, so we shouldn't apply the logic from the parent method, but we also shouldn't throw an error as we did before to be compatible with [Python Database API Specification v2.0](https://www.python.org/dev/peps/pep-0249/#commit).

**Review by vvysotskyi — COMMENTED** [review]
@luocooong, thanks for the review, I've addressed your comments, please take a look.

**Review by luocooong — APPROVED** [review]
@vvysotskyi Nice work. thanks

**Pull request merged**